apps/src/templates/projects/SectionProjectsList.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import ProjectsList from './ProjectsList';
+import _ from 'lodash';
+import color from "../../util/color";
+
+const ALL_STUDENTS = '_all_students';
+
+const styles = {
+  filterDropdown: {
+    margin: 0,
+    color: 'dimgray'
+  },
+  filterRow: {
+    backgroundColor: color.teal,
+    borderBottom: 'solid 1px white',
+    color: color.white,
+    padding: 10,
+    fontSize: 14
+  },
+  filterSpan: {
+    float: 'right',
+    display: 'inline-flex',
+    alignItems: 'center',
+  },
+  clearDiv: {
+    clear: 'both'
+  }
+};
+
+class SectionProjectsList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const studentNames = this.getStudentNames(props.projectsData);
+
+    this.state = {
+      studentNames,
+      selectedStudent: ALL_STUDENTS,
+    };
+  }
+
+  getStudentNames(projectsData) {
+    const studentsMap = {};
+    projectsData.forEach(project => {
+      studentsMap[project['studentName']] = true;
+    });
+    return Object.keys(studentsMap);
+  }
+
+  onChangeStudent(event) {
+    const selectedStudent = event.target.value;
+    this.setState({selectedStudent});
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (_.isEqual(nextProps, this.props)) {
+      return;
```

Comment:
I'm not sure, and I don't think this logic would ever prevent a re-render. I'm calling this a premature optimization and I've removed the early return here. 

Empirical evidence suggests this component never actually has its props changed, because the teacher dashboard re-renders it if we go to another tab and then come back. If we some day move all of teacher dashboard into react and notice a performance problem when this tab updates, we can revisit.